docker/Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM firedrakeproject/firedrake:2024-05 AS spyro_base
+
+USER root
+RUN apt-get update \
+    && apt-get -y dist-upgrade \
+    && apt-get -y install gmsh patchelf \
+    && rm -rf /var/lib/apt/lists/*
+
+USER firedrake
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install wheel --upgrade
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install scipy
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install roltrilinos
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install ROL
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install matplotlib
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install segyio
+
+# spyro dependencies
+USER root
+# Is there a reason for using development versions?
+RUN apt-get update && apt-get install -y libgmp3-dev libmpfr-dev libcgal-dev python3-tk
+USER firedrake
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install --no-dependencies SeismicMesh[io]
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install pyamg
+RUN . /home/firedrake/firedrake/bin/activate; pip3 install meshio
+
+FROM spyro_base AS spyro_release
+
+RUN . /home/firedrake/firedrake/bin/activate; pip install git+https://github.com/Olender/spyro-1.git
```

Comment:
I think it would make sense that the base package also has access to the spyro repository with a git clone and a pip install. That way a new user has access to the notebook_tutorials and can test if everything installed correctly. It is also important to have a shots, results, and meshes folders, since those are the default locations for files generated during simulations (they would already be there with a git clone).

Reply:
During development, I believe the user will want to use his local copy of spyro, so adding the main spyro branch to the image may led to confusions regarding which version is being used.

For testing, the idea is to make the runner clone the commit under test, build the latest test image (to make sure that the runner is using a compatible image), and run the tests. In this case, I think only the spyro under test is important for the image.

These are the reasons why I didn't installed spyro in the base image. The last line of the Dockerfile makes spyro available for the user (in the current branch of the user). Do you think that the current version is enough for accessing notebooks, shots, results, etc?

Reply:
Actually, the test image is always testing the main branch. I have to figure out a way to make it test the current version.

Reply:
Maybe we could have a base version prior to spyro_base with all dependencies installed but no spyro. Then we git clone and checkout outside of the docker before running the tests?